jgrapht-core/src/test/java/org/jgrapht/GraphTestsTest.java

```diff
@@ -0,0 +1,497 @@
+        Graphs.addAllVertices(g, Arrays.asList(1, 2, 3, 4, 5, 6));
+        g.addEdge(1, 2);
+        g.addEdge(2, 3);
+        g.addEdge(3, 1);
+        g.addEdge(4, 5);
+        g.addEdge(5, 6);
+        g.addEdge(6, 4);
+        IntStream.rangeClosed(1, 6).forEach(i -> g.addEdge(i, i));
+        Assert.assertFalse(GraphTests.isEulerian(g));
+    }
+
+    @Test
+    public void testUndirectedEulerian7()
+    {
+        // complete graph of 5 vertices
+        UndirectedGraph<Integer, DefaultEdge> g = new SimpleGraph<>(DefaultEdge.class);
+        CompleteGraphGenerator<Integer, DefaultEdge> gen = new CompleteGraphGenerator<>(5);
+        gen.generateGraph(g, new IntegerVertexFactory(), null);
+        Assert.assertTrue(GraphTests.isEulerian(g));
+    }
+
+    @Test
+    public void testDirectedEulerian1()
+    {
+        // complete graph of 6 vertices
+        DirectedGraph<Integer, DefaultEdge> g1 = new SimpleDirectedGraph<>(DefaultEdge.class);
+        CompleteGraphGenerator<Integer, DefaultEdge> gen1 = new CompleteGraphGenerator<>(6);
+        gen1.generateGraph(g1, new IntegerVertexFactory(), null);
+        Assert.assertTrue(GraphTests.isEulerian(g1));
+
+        // complete graph of 7 vertices
+        DirectedGraph<Integer, DefaultEdge> g2 = new SimpleDirectedGraph<>(DefaultEdge.class);
+        CompleteGraphGenerator<Integer, DefaultEdge> gen2 = new CompleteGraphGenerator<>(7);
+        gen2.generateGraph(g2, new IntegerVertexFactory(), null);
+        Assert.assertTrue(GraphTests.isEulerian(g2));
+    }
+
+    @Test
+    public void testDirectedEulerian2()
+    {
+        DirectedGraph<Integer, DefaultEdge> g = new DirectedPseudograph<>(DefaultEdge.class);
+        Graphs.addAllVertices(g, Arrays.asList(1));
+        Assert.assertTrue(GraphTests.isEulerian(g));
+        g.addEdge(1, 1);
+        g.addEdge(1, 1);
+        Assert.assertTrue(GraphTests.isEulerian(g));
+        Graphs.addAllVertices(g, Arrays.asList(2));
+        g.addEdge(2, 1);
+        Assert.assertFalse(GraphTests.isEulerian(g));
+    }
+
+    @Test
+    public void testDirectedEulerian3()
+    {
+        DirectedGraph<Integer, DefaultEdge> g = new DirectedPseudograph<>(DefaultEdge.class);
+        Graphs.addAllVertices(g, Arrays.asList(1, 2, 3, 4, 5));
+        Assert.assertFalse(GraphTests.isEulerian(g));
+        g.addEdge(1, 2);
+        g.addEdge(2, 3);
+        g.addEdge(3, 4);
+        g.addEdge(4, 5);
+        g.addEdge(5, 1);
+        Assert.assertTrue(GraphTests.isEulerian(g));
+        g.addEdge(2, 1);
+        g.addEdge(3, 2);
+        g.addEdge(4, 3);
+        g.addEdge(5, 4);
+        Assert.assertFalse(GraphTests.isEulerian(g));
+        g.addEdge(1, 1);
+        g.addEdge(2, 2);
+        g.addEdge(3, 3);
+        g.addEdge(4, 4);
+        g.addEdge(5, 5);
+        Assert.assertFalse(GraphTests.isEulerian(g));
+        g.addEdge(1, 5);
+        Assert.assertTrue(GraphTests.isEulerian(g));
+    }
+
+    class IntegerVertexFactory
```

Comment:
I added a default implementation called `IntegerVertexFactory` but kept it in the test classpath.